### PR TITLE
Resource module for IOS ACL

### PR DESCRIPTION
**Pull request by justjais**
##### SUMMARY

Resource module for IOS ACL, ref: [ios_acl](https://github.com/ansible-network/resource_module_models/pull/60)

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

ios_acl

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/network/ios/ios_acl.py [hunk 1]
```diff
@@ -0,0 +1,946 @@
...
+    """
+    Main entry point for module execution
+
+    :returns: the result form module invocation
+    """
+    required_if = [('state', 'merged', ('config',)),
+                   ('state', 'replaced', ('config',)),
+                   ('state', 'overridden', ('config',))]
+
+    module = AnsibleModule(argument_spec=AclArgs.argument_spec,
```
Comment (samdoran) [inline lib/ansible/modules/network/ios/ios_acl.py:939]: `AclArgs.argument_spec` is returning `None`, which is causing `validate-modules` to stack trace. There's probably a bug in that class.

Comment (justjais) [reply lib/ansible/modules/network/ios/ios_acl.py:939]: @samdoran Thanks for reviewing and fixing the stack trace error, and in this case doc was returning None coz of the wrong indentation in the doc.
So, stack trace actually helped in triaging the issue and in the fix I can see you have returned empty dict if None which might skip the error scenarios, can we include any warnings for the users in this case when we encounter None.

Comment (samdoran) [reply lib/ansible/modules/network/ios/ios_acl.py:939]: We could probably add another check for the argument spec being `None`.

Comment (justjais) [reply lib/ansible/modules/network/ios/ios_acl.py:939]: yes, that'd be a better fix for the same.

**Comment by justjais** [comment]
Closing this PR as opening a new PR with the same changes #66629

**Pull request closed without merging**